partner_deduplicate_by_website/tests/test_crm_deduplicate_by_website.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2016 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Tecnativa - Cristina Martin
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+from odoo.tools.safe_eval import safe_eval
+
+
+class TestDeduplicateByWebsite(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner_1 = cls.env['res.partner'].with_context(
+            tracking_disable=True).create({
+                'name': 'Partner 1',
+                'website': 'www.test-deduplicate.com',
+                'email': 'test@deduplicate.com',
+            })
+
+        cls.partner_2 = cls.env['res.partner'].with_context(
+            tracking_disable=True).create({
+                'name': 'Partner 2',
+                'website': 'www.test-deduplicate.com',
```

Comment:
Like @rafaelbn says, it would be good to improve a little bit the detection. All these should be considered the same:

- www.test-deduplicate.com
- test-deduplicate.com
- http://test-deduplicate.com
- https://test-deduplicate.com
- http://test-deduplicate.com/
- https://test-deduplicate.com/
- http://www.test-deduplicate.com
- https://www.test-deduplicate.com
- http://www.test-deduplicate.com/
- https://www.test-deduplicate.com/

You can try test-driven development here if you want. Just add all the partners with these variants and the expected result, and let the test guide you. :blush: 

OTOH, this PR is just a migration. As a migration, improving this part shouldn't be a requirement. If you prefer, just add this note to the roadmap and someone can do it later as the need arises.

Reply:
Yes, this is totally out of scope of the migration, and it's a feature not needed for the customer. Document it on the ROADMAP with the detail put by @Yajo, but nothing more.